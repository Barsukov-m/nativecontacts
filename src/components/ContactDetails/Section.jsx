import { View } from 'react-native';
import styles from './styles';

const Section = ({ children, ...rest }) => {
  return (
    <View style={styles.section} {...rest}>
      {children}
    </View>
  );
};

export default Section;
