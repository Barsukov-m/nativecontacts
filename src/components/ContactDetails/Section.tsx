import React from 'react';
import { View, ViewProps } from 'react-native';
import styles from './styles';

interface SectionProps extends ViewProps {
  children?: React.ReactNode;
}

const Section: React.FC<SectionProps> = ({ children, ...rest }) => {
  return (
    <View style={styles.section} {...rest}>
      {children}
    </View>
  );
};

export default Section;
