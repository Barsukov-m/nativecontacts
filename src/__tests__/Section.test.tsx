import { Text } from 'react-native';
import { render } from '@testing-library/react-native';
import Section from '../components/ContactDetails/Section';

describe('<Section />', () => {
	it('renders correctly', () => {
		const tree = render(<Section testID="contact-section" />);
		const { getByTestId } = tree;
		const sectionComponent = getByTestId('contact-section');

		expect(sectionComponent).toBeTruthy();

		expect(tree).toMatchSnapshot();
	});

	it('renders child components', () => {
		const { getByText } = render(
			<Section>
				<Text>Child Content</Text>
			</Section>
		);
		const childText = getByText('Child Content');

		expect(childText).toBeTruthy();
	});
});
