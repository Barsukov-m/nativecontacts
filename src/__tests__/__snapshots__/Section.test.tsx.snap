// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> renders correctly 1`] = `
<View
  style={
    {
      "borderBottomColor": "#eee",
      "borderBottomWidth": 1,
    }
  }
  testID="contact-section"
/>
`;
